internal/cli/opsmanager/logs/decrypt.go

```diff
@@ -0,0 +1,126 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package logs
+
+import (
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/mongodb/mongocli/internal/cli"
+	"github.com/mongodb/mongocli/internal/decryption"
+	"github.com/mongodb/mongocli/internal/flag"
+	"github.com/mongodb/mongocli/internal/usage"
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+)
+
+type DecryptOpts struct {
+	cli.GlobalOpts
+	cli.OutputOpts
+	cli.DownloaderOpts
+	inFileName                    string
+	kmipServerCAFileName          string
+	kmipClientCertificateFileName string
+	localKeyFileName              string
+}
+
+func (opts *DecryptOpts) initFiles() func() error {
+	// Validate the provided files can be open
+	return func() error {
+		return nil
+	}
+}
+
+func (opts *DecryptOpts) stdOutMode() bool {
+	return opts.Out == ""
+}
+
+func (opts *DecryptOpts) Run() error {
+	var outWriter io.WriteCloser
+
+	if opts.stdOutMode() {
+		outWriter = os.Stdout
+	} else {
+		var err error
+		outWriter, err = opts.NewWriteCloser()
+		if err != nil {
+			return err
+		}
+		defer outWriter.Close()
+	}
+
+	inReader, err := opts.Fs.Open(opts.inFileName)
+	if err != nil {
+		return err
+	}
+	defer inReader.Close()
+
+	keyProviderOpts := decryption.KeyProviderOpts{
+		LocalKeyFileName:              opts.localKeyFileName,
+		KMIPServerCAFileName:          opts.kmipServerCAFileName,
+		KMIPClientCertificateFileName: opts.kmipClientCertificateFileName,
+	}
+
+	if err := decryption.Decrypt(inReader, outWriter, keyProviderOpts); err != nil {
+		if !opts.stdOutMode() {
+			_ = opts.OnError(outWriter)
+			return err
+		}
+	}
+
+	if !opts.stdOutMode() {
+		fmt.Printf("Decrypt of %s to %s completed.\n", opts.inFileName, opts.Out)
+	}
+
+	return nil
+}
+
+// mongocli om logs decrypt --localKey <localKeyFile> --kmipServerCAFile <caFile> –-kmipClientCertificateFile <certFile> --file <encryptedLogFile> --out <outputLogFile>.
+func DecryptBuilder() *cobra.Command {
+	opts := &DecryptOpts{}
+	opts.Fs = afero.NewOsFs()
+	cmd := &cobra.Command{
+		Use:   "decrypt",
+		Short: "Decrypts a log file with the provided local key file or KMIP files.",
+		Example: `
+  $ mongocli ops-manager logs decrypt --localKeyFile /path/to/keyFile --file /path/to/logFile.bson --out /path/to/file.json
+  $ mongocli ops-manager logs decrypt --localKeyFile /path/to/keyFile --file /path/to/logFile.json --out /path/to/file.json`,
+		PreRunE: func(cmd *cobra.Command, args []string) error {
+			return opts.PreRunE(opts.initFiles())
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return opts.Run()
+		},
+	}
+
+	cmd.Flags().StringVarP(&opts.inFileName, flag.File, flag.FileShort, "", usage.EncryptedLogFile)
+	cmd.Flags().StringVarP(&opts.Out, flag.Out, "", "", usage.OutputLogFile)
```

Comment:
[q] should this be
```suggestion
	cmd.Flags().StringVarP(&opts.Out, flag.Out, "o", "", usage.OutputLogFile)
```

Reply:
no, currently `o` is for format like `-o json` this is out for file output like `--out /path/to/file.json`

Reply:
> no, currently o is for format like -o json this is out for file output like --out /path/to/file.json
close but no, 
--out|-o both work with `json`, go template or json path (usually)

Reply:
see Andrea's comment you don't need --output|-o as you don't use `opts.Print()`

Reply:
I removed the flag.Output (which was added by mistake) and for this command we'll have --out|-o